tests/unit_tests/levin.cpp

```diff
@@ -504,44 +550,65 @@ TEST_F(levin_notify, fluff_with_padding)
     }
     notifier.new_out_connection();
     io_service_.poll();
-    {
-        const auto status = notifier.get_status();
-        EXPECT_FALSE(status.has_noise);
-        EXPECT_FALSE(status.connections_filled); // not tracked
-    }
 
     std::vector<cryptonote::blobdata> txs(2);
     txs[0].resize(100, 'f');
     txs[1].resize(200, 'e');
 
+    std::vector<cryptonote::blobdata> sorted_txs = txs;
+    std::sort(sorted_txs.begin(), sorted_txs.end());
+
     ASSERT_EQ(10u, contexts_.size());
+    bool has_stemmed = false;
+    bool has_fluffed = false;
+    while (!has_stemmed || !has_fluffed)
     {
         auto context = contexts_.begin();
-        EXPECT_TRUE(notifier.send_txs(txs, context->get_id()));
+        EXPECT_TRUE(notifier.send_txs(txs, context->get_id(), events_, cryptonote::relay_method::stem));
 
         io_service_.reset();
         ASSERT_LT(0u, io_service_.poll());
-        notifier.run_fluff();
-        ASSERT_LT(0u, io_service_.poll());
+        const bool is_stem = events_.has_stem_txes();
+        EXPECT_EQ(txs, events_.take_relayed(is_stem ? cryptonote::relay_method::stem : cryptonote::relay_method::fluff));
 
+        if (!is_stem)
+        {
+            notifier.run_fluff();
+            ASSERT_LT(0u, io_service_.poll());
+        }
+
+        std::size_t send_count = 0;
         EXPECT_EQ(0u, context->process_send_queue());
         for (++context; context != contexts_.end(); ++context)
-            EXPECT_EQ(1u, context->process_send_queue());
+        {
+            const std::size_t sent = context->process_send_queue();
+            if (sent && is_stem)
```

Comment:
add braces to this if clause to avoid the annoying gcc warning _warning: suggest explicit braces to avoid ambiguous ‘else’ [-Wdangling-else]_

Reply:
Did this get printed when you tried to compile? This isn't nested `if` explicitly within our code, nor does it appear to be a nested `if` when the macro is expanded. I suppose its worth doing either way - this is in the "upstream" Dandelion++ PR not this PR though.